File: C#.Net/JDI/Web/JDI Web/Selenium/Elements/Common/Image.cs
using System;
using JDI_Core.Interfaces.Common;
using JDI_Web.Selenium.Base;
using JDI_Web.Selenium.Elements.Base;
using OpenQA.Selenium;

namespace JDI_Web.Selenium.Elements.Common
{
    public class Image : Clickable, IImage
    {
        public Image() : this (null) { }
        public Image(By byLocator = null, IWebElement webElement = null)
            : base(byLocator, webElement)
        { }

        protected Func<WebBaseElement, string> GetSourceFunc = 
            el => el.WebAvatar.FindImmediately(() => el.WebElement.GetAttribute("src"), "");
        public string GetSource()
        {
            return Invoker.DoJActionResult("Get image source for Element "+ this, GetSourceFunc);
        }

        protected Func<WebBaseElement,string> GetAltFunc =
            el => el.WebAvatar.FindImmediately(() => el.WebElement.GetAttribute("alt"), "");
        public string GetAlt()
        {
            return Invoker.DoJActionResult("Get image title for Element "+ this, GetAltFunc);
        }

    }
}

File: Python/JDI/web/selenium/elements/common/image.py
from JDI.core.utils.decorators import scenario
from JDI.web.selenium.elements.base.clickable import Clickable


class Image(Clickable):

    def __init__(self, by_locator=None):
        super(Image, self).__init__(by_locator)

    @scenario(action_name="Get image source for Element")
    def get_source(self):
        return self.get_web_element().get_attribute("src")

    @scenario(action_name="Get image title for Element")
    def get_alt(self):
        return self.get_web_element().get_attribute("alt")